api/numaresourcesoperator/v1/helper/nodegroup/nodegroup_test.go

```diff
@@ -203,9 +244,9 @@ func TestFindTrees(t *testing.T) {
 			if err != nil {
 				t.Errorf("unexpected error checking backward compat: %v", err)
 			}
-			compatNames := mcpNamesFromList(gotMcps)
-			if !reflect.DeepEqual(gotNames, compatNames) {
-				t.Errorf("Trees mismatch (non backward compatible): got=%v compat=%v", gotNames, compatNames)
+			compatibleNames := mcpNamesFromList(gotMcps)
+			if !containsSliceItems(gotNames, compatibleNames) {
```

Comment:
You can use `set.Set` for finding the matches. 
```
A := set.New(gotNames...)
if !A.HasAll(compatibleNames...)
...
```

Reply:
This indeed looks better and shorter, if I understand correctly this is the set util you are referring to:
https://github.com/kubernetes/client-go/blob/v1.5.2/1.5/pkg/util/sets/string.go#L69 
I see that ultimately the method is using similar principles of map[string]struct{}, and my question is whether we want to add a dependency or reimplement this as suggested in this PR? 
I'm still working on how to identify in which cases we need to add a dependency and in which we would go with reimplementation, but for this I can say that is not required in other places in the project and the implementation and testing is very simple so why to increase our dependencies?.
Please share your thoughts on this I'd like to learn more.

Reply:
the question is very correct, but in this case we already depend on `client-go`, and we use sets already in the codebase. You can use `k8s.io/apimachinery/pkg/util/sets`, same reasoning. please see 
`pkg/validator/validator.go` and `internal/schedcache/synced.go` for example.

Reply:
> I'm still working on how to identify in which cases we need to add a dependency and in which we would go with reimplementation, but for this I can say that is not required in other places in the project and the implementation and testing is very simple so why to increase our dependencies?

Unfortunately there's no hard rule and ideas vary wildly. Everyone has to make their own mind.
Few questions may give some ideas
1. is the new proposed dep part of a module we pull already? (e.g. apimachinery/pkg/util/set while we already depend on apimachinery)
2. how happy are we with the same module? are we happy to stay with that dep or we have a general sentiment to migrate? (if not completely happy let's first stop adding, so the transition is easier)
3. how widespread is that new dep? (e.g. in the k8s ecosystem world everyone is using it, so all other factors being equal using the same code makes our project easier to understand because it looks like the others)
4. (from now on the usual suspects) how hard is to reimplement that dep? if we use 2 functions but these 2 functions are heavily optimized compression or crypto algos, well :)
5. how heavy (code size) is the new dep? Do we need to pull in 10 megs and 100k LoCs just for 2 functions?
6. how well maintained is the new dep?

Reply:
Thanks @ffromani for the tips that should be handy going forward to help me decide in similar situation.
I'll be using the apimachinary packages in this case, thank you for the references.

Reply:
adjusted in the latest updates. Thanks!